### PR TITLE
Labs 74 mysql auto subscriber provider

### PR DESCRIPTION
**Pull request by vokoblin**
Getting all the existent subscribers.

### CODE REVIEW

File: src/AutoSubscribe/Entity/User.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * User: Volodymyr
+ * Date: 03/09/2015
+ * Time: 12:16
+ */
+class User implements IUser
+{
+    /**
+     * @var
+     */
+    private $userId;
```
Comment (MeliaBurnley) [inline src/AutoSubscribe/Entity/User.php:18]: Type

File: test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        $this->connection->rollBack();
+    }
+
+    /**
+     * @test
+     */
+    public function duringConstructionOfGateway_GatewayIsNotNull()
+    {
+        $this->assertNotNull( $this->gateway );
+    }
```
Comment (MeliaBurnley) [inline test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php]: I'm a little confused as to why you need this?

File: test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * @test
+     */
+    public function givenNoSuscribers_ReturnEmptyArray()
```
Comment (MeliaBurnley) [inline test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php]: "Suscribers" -> "Subscribers"

File: test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * @test
+     */
+    public function givenExistentSubscribers_whenGettingCallingGetID_ReturnsCorrectId()
+    {
+        $subscriberName = "TheSubscribes";
+        $subscriber = new User( $subscriberName );
+        $this->assertEquals( $subscriberName, $subscriber->getId() );
+    }
```
Comment (MeliaBurnley) [inline test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php]: Also unsure as to what this is for, surely that's testing User and should be in a separate test?

File: db/3.sql [hunk 5]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE `subscribers` (
```
Comment (MeliaBurnley) [inline db/3.sql]: You're gonna want to stick this in the labs-schema repo now

File: src/AutoSubscribe/Entity/User.php [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+use Clearbooks\Labs\AutoSubscribe\Entity\User as IUser;
+
+/**
+ * Created by PhpStorm.
+ * User: Volodymyr
+ * Date: 03/09/2015
+ * Time: 12:16
+ */
+class User implements IUser
```
Comment (danielburnley) [inline src/AutoSubscribe/Entity/User.php]: Is there a reason we have so many user classes that all do the exact same things?

File: test/AutoSubscribe/Entity/UserTest.php [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        $userData[ 'userId' ] = self::USER_ID;
+
+        $this->assertUserGettersMatchExpectedUserData( $user, $userData );
+    }
+
+    /**
+     * @param User $user
+     * @param array $emptyUserData
+     */
+    private function assertUserGettersMatchExpectedUserData( User $user, $emptyUserData )
```
Comment (MeliaBurnley) [inline test/AutoSubscribe/Entity/UserTest.php]: Should be up above setUp

**Pull request merged**